src/ui/Pager/Pager.ts

```diff
@@ -225,10 +246,25 @@ export class Pager extends Component {
   }
 
   private handleNoResults(data: INoResultsEventArgs) {
-    var lastValidPage = this.computePagerBoundary(data.results.totalCountFiltered, data.results.totalCount).lastResultPage;
-    if (this.currentPage > lastValidPage) {
-      this.ignoreNextQuerySuccess = true;
-      this.setPage(lastValidPage);
+    this.ignoreNextQuerySuccess = true;
+    let lastValidPage;
+    if (data.results.totalCount > 0) {
+      // First scenario : The index returned less results than expected (because of folding).
+      // Recalculate the last valid page, and change to that new page.
+      const possibleValidPage = this.computePagerBoundary(data.results.totalCountFiltered, data.results.totalCount).lastResultPage;
+      if (this.currentPage > possibleValidPage) {
+        lastValidPage = possibleValidPage;
+      }
+    } else if (this.currentPage > this.getMaxNumberOfPagesForCurrentResultsPerPage()) {
+      // Second scenario : Someone tried to access a non-valid page by the URL for example, which is  higher than the current possible with the number of
```

Comment:
An ogre ran away with the end of this comment. Get it back! 👹 :crossed_swords:  🏃